app/code/Magento/Store/App/Response/Redirect.php

```diff
@@ -7,10 +7,21 @@
  */
 namespace Magento\Store\App\Response;
 
+use Magento\Framework\App\Config\ScopeConfigInterface;
 use Magento\Store\Api\StoreResolverInterface;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\Store;
 
 class Redirect implements \Magento\Framework\App\Response\RedirectInterface
 {
+    const XML_PATH_USE_CUSTOM_ADMIN_PATH = 'admin/url/use_custom_path';
```

Comment:
Again, `Backend` module depends on `Store` and not vise versa, `Store` module must not receive any knowledge about admin part. I cannot say a word about whether proposed implementation is correct or not, just pointing out to a modularity violation.

It may be a good idea to wait for a full code review before doing further changes.

Reply:
@briscoda another small note, there is no need to create new pull requests, just continue work with the same branch, you can force push at any moment also.

Reply:
I appreciate your notes, thank you. There is also a really simple solution to this too, which would be to just call:

$this->_storeManager->getStore(0);

And explicitly pass it 0 which forces it to return the configured admin domain. It's like a one-line thing which could be inserted into the _isUrlInternal function.

Just for curiosity though, in this current implementation, it calls information from the scopeConfig which is in Framework. So although these constants use config paths which have been set for the admin, I'm not sure if they're strictly tied to the Backend module. But I see what you're saying.

Reply:
Alternatively, maybe a plugin that lived in the Backend module that happened after the isUrlInternal function could accomplish this as well. That way references to the admin paths are not contained in this store module but live there instead.